modules/base/rendering/screenspaceinsetblackout.cpp

```diff
@@ -0,0 +1,768 @@
+            pointRemoved = true;
+        });
+        addProperty(removeSelector);
+        addProperty(removeButton);
+        for (int i = 0; i < points.size(); ++i) {
+            removeSelector.addOption(i, std::format("Point #{}", i + 1));
+        }
+    }
+
+}
+
+void ScreenSpaceInsetBlackout::BlackoutShape::Spline::addPoint() {
+    // Add new position to raw data
+    const int index = addSelector;
+    data.insert(data.begin() + index, newPointPosition);
+}
+
+void ScreenSpaceInsetBlackout::BlackoutShape::Spline::removePoint() {
+    // Remove raw data for the deleted point
+    const int indexToRemove = removeSelector;
+    data.erase(data.begin() + indexToRemove);
+ }
+
+ScreenSpaceInsetBlackout::BlackoutShape::Corners::Corners(std::vector<glm::vec2>& inData)
+    : PointOwner(inData, "Corners", "Corners")
+{
+    // Create corner Points (TopLeft -> TopRight -> BottomRight -> BottomLeft)
+    points.push_back(std::make_unique<Point>(data[0], "TopLeft", "Top-Left"));
+    points.push_back(std::make_unique<Point>(data[1], "TopRight", "Top-Right"));
+    points.push_back(std::make_unique<Point>(data[2], "BottomRight", "Bottom-Right"));
+    points.push_back(std::make_unique<Point>(data[3], "BottomLeft", "Bottom-Left"));
+
+    // OnChange functions and add to GUI
+    for (int i = 0; i < 4; i++) {
+        points[i]->prop->onChange([this, i]() {
+            points[i]->updateData();
+            dataHasChanged = true;
+        });
+        addProperty(points[i]->prop.get());
+    }
+}
+
+ScreenSpaceInsetBlackout::BlackoutShape::BlackoutShape(const ghoul::Dictionary& dictionary)
+    : properties::PropertyOwner({ "BlackoutShape", "Blackout Shape", "" })
+    , enableCalibrationColor(CalibrationColorInfo, false)
+    , enableCalibrationPattern(CalibrationPatternInfo, false)
+    , calibrationTexturePath(CalibrationTextureInfo)
+    , copyToClipboardTrigger(CopyToClipboardInfo)
+{
+    const Parameters params = codegen::bake<Parameters>(dictionary);
+
+    // Stores all position data read from asset file
+    cornerData = params.blackoutshape.corners;
+    topSplineData = params.blackoutshape.top.value_or(std::vector<glm::vec2>());
+    rightSplineData = params.blackoutshape.right.value_or(std::vector<glm::vec2>());
+    bottomSplineData = params.blackoutshape.bottom.value_or(std::vector<glm::vec2>());
+    leftSplineData = params.blackoutshape.left.value_or(std::vector<glm::vec2>());
+
+    // Create sections
+    corners = new Corners(cornerData);
+    topSpline = new Spline(topSplineData, "Top");
+    rightSpline = new Spline(rightSplineData, "Right");
+    bottomSpline = new Spline(bottomSplineData, "Bottom");
+    leftSpline = new Spline(leftSplineData, "Left");
+
+    // Add to GUI
+    addPropertySubOwner(corners);
+    addPropertySubOwner(topSpline);
+    addPropertySubOwner(rightSpline);
+    addPropertySubOwner(bottomSpline);
+    addPropertySubOwner(leftSpline);
+
+    // Add additional controls to GUI
```

Comment:
For consistency with other renderables
```suggestion
    const Parameters p = codegen::bake<Parameters>(dictionary);

    cornerData = p.blackoutshape.corners;
    corners = new Corners(cornerData);
    addPropertySubOwner(corners);
    
    topSplineData = p.blackoutshape.top.value_or(std::vector<glm::vec2>());
    topSpline = new Spline(topSplineData, "Top");
    addPropertySubOwner(topSpline);

    rightSplineData = p.blackoutshape.right.value_or(std::vector<glm::vec2>());
    rightSpline = new Spline(rightSplineData, "Right");
    addPropertySubOwner(rightSpline);

    bottomSplineData = p.blackoutshape.bottom.value_or(std::vector<glm::vec2>());
    bottomSpline = new Spline(bottomSplineData, "Bottom");
    addPropertySubOwner(bottomSpline);

    leftSplineData = p.blackoutshape.left.value_or(std::vector<glm::vec2>());
    leftSpline = new Spline(leftSplineData, "Left");
    addPropertySubOwner(leftSpline);
```